src/main/java/org/embulk/output/snowflake/SnowflakeCopyBatchInsert.java

```diff
@@ -351,32 +351,31 @@ public UploadTask(
     }
 
     public Void call() throws IOException, SQLException, InterruptedException {
-      logger.info(
-          String.format(
-              "Uploading file id %s to Snowflake (%,d bytes %,d rows)",
-              snowflakeStageFileName, file.length(), batchRows));
-
       int retries = 0;
       try {
         long startTime = System.currentTimeMillis();
         // put file to snowflake internal storage
         SnowflakeOutputConnection con = (SnowflakeOutputConnection) connector.connect(true);
 
-        FileInputStream fileInputStream = new FileInputStream(file);
         do {
           try {
+            logger.info(
+                String.format(
+                    "Uploading file id %s to Snowflake (%,d bytes %,d rows)",
+                    snowflakeStageFileName, file.length(), batchRows));
+            FileInputStream fileInputStream = new FileInputStream(file);
             con.runUploadFile(stageIdentifier, snowflakeStageFileName, fileInputStream);
+            break;
           } catch (SQLException e) {
             retries++;
-            if (retries > this.maxUploadRetries) {
+            if (retries >= this.maxUploadRetries) {
```

Comment:
When `maxUploadRetries` is 1, we will expect to retry once.
But this may not be retried even once.

Reply:
fixed
- https://github.com/trocco-io/embulk-output-snowflake/pull/56/commits/279c0bf9308f49dead68d189d161e1d99ae4fd74